utils.sh

```diff
@@ -185,3 +185,19 @@ function bmo_config_map {
     
     cp ocp/deploy/metal3-config.yaml assets/generated/99_metal3-config.yaml
 }
+
+function image_mirror_config {
+    if [ ! -z "${MIRROR_IMAGES}" ]; then
+        TAGGED=$(echo $OPENSHIFT_RELEASE_IMAGE | sed -e 's/release://')
+        RELEASE=$(echo $OPENSHIFT_RELEASE_IMAGE | grep -o 'registry.svc.ci.openshift.org[^":]\+')
```

Comment:
This doesn't always work, that's what I got while using nightly
```bash
Success
Update image:  192.168.123.141:5000/localimages/local-release-image:4.3.0-0.nightly-2019-11-25-022933
Mirror prefix: 192.168.123.141:5000/localimages/local-release-image

To use the new mirrored repository to install, add the following section to the install-config.yaml:

imageContentSources:
- mirrors:
  - 192.168.123.141:5000/localimages/local-release-image
  source: quay.io/openshift-release-dev/ocp-v4.0-art-dev
- mirrors:
  - 192.168.123.141:5000/localimages/local-release-image
  source: registry.svc.ci.openshift.org/ocp/release
```

Reply:
Maybe if the user wants to mirror images, we should ask them for the exact sources? What do you think?

Reply:
I can confirm that was the behavior I got as well using 4.3 nightly's on an environment where I was doing disconnected.

Reply:
I don't like how the data structure is laid out at all...  What we have here is a hodge podge of array and string objects.  It will be mind bending to code against this.

How about ...
```
imageContentSources:
-
  target: <url>:<port> 
  user: <user ie. openshift-release-dev or localimages>
  release: <release ie. ocp-v4.0-art-dev or release>
-
  target: <url>:<port> 
  user: <user ie. openshift-release-dev or localimages>
  release: <release ie. ocp-v4.0-art-dev or release>
-
  target: <url>:<port> 
  user: <user ie. openshift-release-dev or localimages>
  release: <release ie. ocp-v4.0-art-dev or release>
```

Reply:
This is an API defined by OpenShift, it's not something we can revisit.

Reply:
@mazzystr I'm not sure what you mean. We're simply using openshift-installer-provided values. We don't define any data structures.